File: Programmers/src/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        // 남은 작업일수를 저장할 큐
        Queue<Integer> q = new LinkedList<>();
        
        for(int i=0; i<speeds.length; i++){
            // 남은 작업일을 계산하여 큐에 추가
            q.offer((100 - progresses[i]) / speeds[i] + ((100 - progresses[i]) % speeds[i] == 0 ? 0 : 1));
        }
        
        // 현재 카운트 1
        int cnt=1;
        // 첫번째 항목 꺼냄
        int now = q.poll();
        // 결과를 담는 배열
        ArrayList<Integer> list = new ArrayList<>();
        
        while(!q.isEmpty()){
            // 현재 작업일과 다른 작업일 비교값
            int next = q.poll();
            // 현재 작업일이 더 많은 경우 cnt++
            if(next<=now)
                cnt++;
            // 현재 작업일이 더 작은 경우
            else {
                // 배열에 계산한 cnt 추가
                list.add(cnt);
                // 현재 작업일 next로 변경
                now = next;
                // cnt 초기화
                cnt=1;
            }
        }
        
        // 마지막 cnt는 추가되지 않아 따로 추가
        list.add(cnt);
        
        // ArrayList를 배열로 변환  
        int[] answer = new int[list.size()];
        for(int i=0; i<answer.length; i++)
            answer[i] = list.get(i);
        
        return answer;
    }
}

File: Programmers/src/Javascript/Lv2/기능개발.js
function solution(progresses, speeds) {
    var answer = [];        // answer: 각 배포마다 배포되는 기능의 수가 적힌 정수 배열
    let days = 1;           // days: 배포일
    let cnt = 0;            // cnt: 오늘 배포되는 기능의 수
    let progress = 0;       // progress: 현재 기능의 작업 진도
    
    // 모든 작업이 다 배포될 때까지 반복
    while(progresses[0]){
        progress = progresses[0] + (speeds[0] * days);      // 첫 번째 기능의 작업 진도
        
        // 첫 번째 기능의 작업 진도가 100 이상인 경우 배포 완료
        if(progress >= 100){
            cnt++;                  // 배포 완료된 기능 개수 추가
            progresses.shift();     // 배포 완료된 작업 제거
            speeds.shift();         // 배포 완료된 작업의 속도 제거
        } else {
            if(cnt > 0)
                answer.push(cnt);   // 배포 완료된 기능이 있는 경우, answer에 push
            
            days++;                 // 배포일 증가 (다음날)
            cnt = 0;                // 배포 완료된 기능 개수 초기화
        }
    }
    
    // 모든 작업이 다 배포되고 나면, 마지막으로 카운트된 배포 완료 기능 개수 push
    answer.push(cnt);
    
    return answer;
}